File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlParameterValue.cs
namespace T1.SqlSharp.Expressions;

public class SqlParameterValue : ISqlExpression
{
    public SqlType SqlType => SqlType.ParameterValue;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_ParameterValue(this);
    }

    public string Name { get; set; } = string.Empty;
    public string Value { get; set; } = string.Empty;
    public string ToSql()
    {
        return $@"{Name}={Value}";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlParameterValue.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlParameterValue implements ISqlExpression {
    SqlType: SqlType = SqlType.ParameterValue;
    Span: TextSpan = new TextSpan();
    Name: string = '';
    Value: string = '';

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_ParameterValue(this);
    }

    ToSql(): string {
        return `${this.Name}=${this.Value}`;
    }
} 